### ISSUE TITLE
MissingMemberException when receiving messages

### ISSUE THREAD

**Issue by SamuelDebruyn** [Azure/azure-service-bus-dotnet#653]
## Actual Behavior
1. Getting thousands of MissingMemberExceptions when calling RegisterMessageHandler 

## Expected Behavior
1. No exceptions

Bug detailed at https://github.com/Azure/azure-amqp/issues/133
Maybe a workaround in this package can be built-in to avoid this?

## Versions  
- NuGet package version or commit ID: 3.1.1
```
=== Visual Studio Enterprise 2017 for Mac ===

Version 7.8.1 (build 4)
Installation UUID: e1324239-09f8-4ef0-841d-818a176fdfbd
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.0.0.0 ( / b40230c0)

	Package version: 518000248

=== Mono Framework MDK ===

Runtime:
	Mono 5.18.0.248 (2018-08/a4956c837e1) (64-bit)
	Package version: 518000248

=== NuGet ===

Version: 4.8.0.5385

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.2.0
	2.1.2
	2.1.1
	2.1.0
	2.0.7
	2.0.5
SDK: /usr/local/share/dotnet/sdk/2.2.100/Sdks
SDK Versions:
	2.2.100
	2.1.302
	2.1.301
	2.1.300
	2.1.200
	2.1.4
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.18.0/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.4
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin.Android ===

Version: 9.1.8.0 (Visual Studio Enterprise)
Android SDK: /Users/sam/Library/Android/sdk
	Supported Android versions:
		2.3   (API level 10)
		4.0.3 (API level 15)
		5.0   (API level 21)
		6.0   (API level 23)
		7.0   (API level 24)
		7.1   (API level 25)
		8.0   (API level 26)
		8.1   (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.1
SDK Build Tools Version: 28.0.3

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/sam/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.9
1.8.0-9
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android Device Manager ===

Version: 7.8.1.0
Hash: 8924ea4a

=== Apple Developer Tools ===

Xcode 10.1 (14460.46)
Build 10B61

=== Xamarin.Mac ===

Version: 5.2.1.15 (Visual Studio Enterprise)
Hash: d60abd198
Branch: 
Build date: 2019-02-01 12:23:30-0500

=== Xamarin.iOS ===

Version: 12.2.1.15 (Visual Studio Enterprise)
Hash: d60abd198
Branch: d15-9
Build date: 2019-02-01 12:23:29-0500

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 708010004
Git revision: 8e2582c0fc1558364b67502b43d2e6a44ff6bc48
Build date: 2019-02-21 18:39:01+00
Build branch: release-7.8
Xamarin extensions: af96e69ddf566b275db40628f8530ca48a97f0b9

=== Operating System ===

Mac OS X 10.14.3
Darwin 18.2.0 Darwin Kernel Version 18.2.0
    Thu Dec 20 20:46:53 PST 2018
    root:xnu-4903.241.1~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

Android Signature Tool 2.1.2
Gist Ide Information 1.1.0
NuGet Package Explorer 0.3
Open With 0.1
NuGet Package Management Extensions 0.15
xUnit.NET 2 testing framework support 0.7.5
```

**Comment by SeanFeldman** [comment]
@SamuelDebruyn looking at this issue, isn't the underlying problem is with microsoft.azure.amqp package? I.e. if https://github.com/Azure/azure-amqp/issues/133 is fixed and released, you could bump microsoft.azure.amqp package reference manually until ASB client is updated?

**Comment by SamuelDebruyn** [comment]
Hi @SeanFeldman, the ticket at https://github.com/Azure/azure-amqp/issues/133 isn't being picked up. That means that we can't use servicebus at all in the meantime on Xamarin. Is there any workaround? Or can you ping a colleague from the AMQP team? @xinchen10 @arturl @dlstucki

**Comment by SeanFeldman** [comment]
@SamuelDebruyn for the record, I'm not a Microsoftie but just a collaborator on the project.

AMQP is a dependency for several clients, not just ASB. Would let it be triaged first.
Hopefully it's commented on soon.

**Comment by SamuelDebruyn** [comment]
Fixed in 2.4.1, I'd suggest to update the nuget dependency so that users don't experience the same issue

**Comment by SeanFeldman** [comment]
Already scheduled and tagged in #654
